checks/generic_prometheus_check.py

```diff
@@ -0,0 +1,63 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+from checks import CheckException
+from checks.prometheus_check import PrometheusCheck
+from checks import AgentCheck
+
+# GenericPrometheusCheck is a class that helps instanciating PrometheusCheck only
+# with YAML configurations. As each check has it own states it maintains a map
+# of all checks so that the one corresponding to the instance is executed
+#
+# Minimal example configuration:
+# instances:
+#   - prometheus_url: http://foobar/endpoint
+#     namespace: "foobar"
+#     metrics:
+#       - bar
+#       - foo
+class GenericPrometheusCheck(AgentCheck):
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self.check_map = {}
+        for instance in instances:
```

Comment:
That does not look very AD-ready. Should you hook into `check` when an `instance` is not in the mapping instead?